lib/matplotlib/contour.py

```diff
@@ -1245,27 +1264,28 @@ def _process_levels(self):
         # (Colorbar needs this even for line contours.)
         self._levels = list(self.levels)
 
+        if self.logscale:
+            lower, upper = 1e-250, 1e250
+        else:
+            lower, upper = -1e250, 1e250
+
         if self.extend in ('both', 'min'):
-            self._levels.insert(0, min(self.levels[0], self.zmin) - 1)
+            self._levels.insert(0, lower)
         if self.extend in ('both', 'max'):
-            self._levels.append(max(self.levels[-1], self.zmax) + 1)
+            self._levels.append(upper)
         self._levels = np.asarray(self._levels)
 
         if not self.filled:
             self.layers = self.levels
             return
 
-        # layer values are mid-way between levels
-        self.layers = 0.5 * (self._levels[:-1] + self._levels[1:])
-        # ...except that extended layers must be outside the
-        # normed range:
-        if self.extend in ('both', 'min'):
-            if self.logscale:
-                self.layers[0] = 1e-150
-            else:
-                self.layers[0] = -1e150
-        if self.extend in ('both', 'max'):
-            self.layers[-1] = 1e150
+        # Layer values are mid-way between levels in screen space.
```

Comment:
"... in data space"?

Reply:
No, it is screen space.  That's why the geometric mean is needed for the log scale.